src/nupic/regions/SDRClassifierRegion.py

```diff
@@ -0,0 +1,502 @@
+
+
+  def clear(self):
+    self._sdrClassifier.clear()
+
+
+  def getAlgorithmInstance(self):
+    """Returns instance of the underlying CLAClassifier algorithm object."""
+    return self._sdrClassifier
+
+
+  def getParameter(self, name, index=-1):
+    """
+    Get the value of the parameter.
+
+    @param name -- the name of the parameter to retrieve, as defined
+            by the Node Spec.
+    """
+    # If any spec parameter name is the same as an attribute, this call
+    # will get it automatically, e.g. self.learningMode
+    return PyRegion.getParameter(self, name, index)
+
+
+  def setParameter(self, name, index, value):
+    """
+    Set the value of the parameter.
+
+    @param name -- the name of the parameter to update, as defined
+            by the Node Spec.
+    @param value -- the value to which the parameter is to be set.
+    """
+    if name == "learningMode":
+      self.learningMode = bool(int(value))
+    elif name == "inferenceMode":
+      self.inferenceMode = bool(int(value))
+    else:
+      return PyRegion.setParameter(self, name, index, value)
+
+
+  @staticmethod
+  def getProtoType():
+    """Return the pycapnp proto type that the class uses for serialization."""
+    return SDRClassifierRegionProto
+
+
+  def writeToProto(self, proto):
+    """Write state to proto object.
+
+    proto: SDRClassifierRegionProto capnproto object
+    """
+    proto.classifierImp = self.classifierImp
+    proto.steps = self.steps
+    proto.alpha = self.alpha
+    proto.verbosity = self.verbosity
+    proto.maxCategoryCount = self.maxCategoryCount
+
+    self._sdrClassifier.write(proto.sdrClassifier)
+
+
+  @classmethod
+  def readFromProto(cls, proto):
+    """Read state from proto object.
+
+    proto: CLAClassifierRegionProto capnproto object
+    """
+    instance = cls()
+
+    instance.classifierImp = proto.classifierImp
+    instance.steps = proto.steps
+    instance.alpha = proto.alpha
+    instance.verbosity = proto.verbosity
+    instance.maxCategoryCount = proto.maxCategoryCount
+
+    instance._sdrClassifier = SDRClassifierFactory.read(proto)
+
+    return instance
+
+
+  def reset(self):
+    pass
+
+
+  def compute(self, inputs, outputs):
+    """
+    Process one input sample.
+    This method is called by the runtime engine.
+    @param inputs -- inputs of the classifier region
+    @param outputs -- outputs of the classifier region
+
+    """
+
+    # This flag helps to prevent double-computation, in case the deprecated 
+    # customCompute() method is being called in addition to compute() called 
+    # when network.run() is called
+    self._computeFlag = True
+
+    # An input can potentially belong to multiple categories. 
+    # If a category value is < 0, it means that the input does not belong to 
+    # that category.
+    categories = [category for category in inputs["categoryIn"]
+                  if category >= 0]
+
+    activeCells = inputs["bottomUpIn"]
+    patternNZ = activeCells.nonzero()[0]
```

Comment:
Why declare a variable just to be used once? Why not just have `patternNZ = inputs["bottomUpIn"].nonzero()[0]`